contrib/NeonVM/neonvm.c

```diff
@@ -707,9 +718,50 @@ void exec_CALLP(struct tagTExecutor *self)
         const char *s = top(self->stack)->string;
         printf("%s\n", s);
         pop(self->stack);
+    } else if (strcmp(func, "array__concat") == 0) {
+        Cell *right = cell_fromCell(top(self->stack)); pop(self->stack);
+        Cell *left  = cell_fromCell(top(self->stack)); pop(self->stack);
+        Cell *a = cell_createArrayCell(left->array_size + right->array_size);
+
+        for (uint32_t i = 0; i < left->array_size; i++) {
+            cell_copyCell(&a->array[i], &left->array[i]);
+        }
+        for (uint32_t i = 0; i < right->array_size; i++) {
+            cell_copyCell(&a->array[i+left->array_size], &right->array[i]);
+        }
+        push(self->stack, a);
+        cell_freeCell(left);
+        cell_freeCell(right);
+    } else if (strcmp(func, "array__toString__number") == 0) {
+        // ToDo: Fix this to use proper string appends!!
+        Cell *a = top(self->stack);
+        char *s = malloc(1024);
+        strcpy(s, "[");
+        for (uint32_t x = 0; x < a->array_size; x++) {
+            if (strlen(s) > 1) {
+                strcat(s, ", ");
+            }
+            strcat(s, number_to_string(a->array[x].number));
+        }
+        strcat(s, "]");
+        pop(self->stack);
+        push(self->stack, cell_fromString(s));
```

Comment:
cell_fromString calls strdup() on the passed string but you're not freeing the one you malloced above.